tests/test_lf_controller.cpp

```
@@ -0,0 +1,230 @@
+                  {.name = "leg_left_4_joint"},  {.name = "leg_left_5_joint"},
+                  {.name = "leg_left_6_joint"},  {.name = "leg_right_1_joint"},
+                  {.name = "leg_right_2_joint"}, {.name = "leg_right_3_joint"},
+                  {.name = "leg_right_4_joint"}, {.name = "leg_right_5_joint"},
+                  {.name = "leg_right_6_joint"}, {.name = "torso_1_joint"},
+                  {.name = "torso_2_joint"},     {.name = "arm_left_1_joint"},
+                  {.name = "arm_left_2_joint"},  {.name = "arm_left_3_joint"},
+                  {.name = "arm_left_4_joint"},  {.name = "arm_left_5_joint"},
+                  {.name = "arm_left_6_joint"},  {.name = "arm_left_7_joint"},
+                  {.name = "arm_right_1_joint"}, {.name = "arm_right_2_joint"},
+                  {.name = "arm_right_3_joint"}, {.name = "arm_right_4_joint"},
+                  {.name = "arm_right_5_joint"}, {.name = "arm_right_6_joint"},
+                  {.name = "arm_right_7_joint"},
+              },
+          .has_free_flyer = true,
+      }),
+  };
+  ASSERT_NE(talos_model_ptr, nullptr);
+
+  auto ctrl = LFController();
+  ctrl.initialize(talos_model_ptr);
+  EXPECT_ANY_THROW({ auto _ = ctrl.compute_control({}, {}); });
+}
+
+TEST_F(LfControllerTest, ComputeControlUnknownJoints) {
+  const auto talos_model_ptr = std::shared_ptr{
+      MakeBuilderFrom({
+          .urdf = GetTalosUrdf(),
+          .joints =
+              {
+                  {.name = "root_joint"},
+                  {.name = "leg_left_1_joint"},
+              },
+          .has_free_flyer = true,
+      }),
+  };
+  ASSERT_NE(talos_model_ptr, nullptr);
+
+  auto ctrl = LFController();
+  ctrl.initialize(talos_model_ptr);
+
+  using linear_feedback_controller_msgs::Eigen::Control;
+  using linear_feedback_controller_msgs::Eigen::Sensor;
+
+  // TODO: Create function arguments with an unknown joint name
+  //  (not within the defined model)
```

This is tested in the robot builder unit-tests already no?

---

No, here I mean a mismatch between the names inside Sensor{} and the ones defined by the model through `initilaize()`

---

Oo! I see!
Arf yes ok we need to check that as well.
We can throw something if this appears.

---

I don't really know how the underlying computation works, it is not necessarily needed to throw, maybe they could simply be ignored ?

Since there is no documentation on the API, I considered it as an error and used the 'EXPECT_THROW' as a default error mechanism but maybe it should assert or even it's well defined and should return a valid output vector.

I'll let you guys decide.

---

It should throw, this undefined behavior.
At least:
- A warning internal_moving_joint \in sensor.joint.name and sensor.joint.name > internal_moving_joint.size()
- A runtime_error or logic_error if not internal_moving_joint \in sensor.joint.name
- otherwise expect_no_throw.

---

idem for control.initial_state.